### PR TITLE
Tests for example.org

### PR DESCRIPTION
**Pull request by renatgalimov**
I want to be sure that our example.org tests work as expected.
@orgtre, do you think that they look correct?

This PR depends on #49

### CODE REVIEW

File: anki-editor-tests.el [hunk 1]
```diff
@@ -168,5 +172,107 @@ Simple note body
...
+				                                  ("Text" . "<p>\nCards of this note will be created in {{c1::Default::which deck?}}\n</p>\n"))
+				                                 nil))
+           ("Deck in entry" . #s(anki-editor-note nil "Cloze" "Languages"
+                                 (("Back Extra" . "<p>\nDeck in entry</p>\n")
+                                  ("Text" . "<p>\nCards of this note will be created in {{c1::Languages::which deck?}}\n</p>\n"))
+                                 nil))
+           ("Raw fields" . #s(anki-editor-note nil "Basic" "Default"
+                                 (("Back" . "<p>\nWith property &lt;code&gt;:ANKI<sub>FORMAT</sub>: nil&lt;/code&gt;, content of the\nfield will be sent to Anki &lt;em&gt;unprocessed&lt;/em&gt;.  You can use\nwhatever Anki supports, like HTML tags.\n&lt;br&gt;\n&lt;br&gt;\nThis property is retrieved with inheritance, meaning that it can be\nset in any ancestor entries or at the top of the file with\n&lt;code&gt;#+PROPERTY: ANKI<sub>FORMAT</sub> nil&lt;/code&gt;, it's also possible to\noverride an outer level nil format with &lt;code&gt;:ANKI<sub>FORMAT</sub>: t&lt;/code&gt;.\n</p>\n")
```
Comment (renatgalimov) [inline anki-editor-tests.el:231]: This looks unusual. I see these <code> blocks have been escaped. I think it's wrong.

Comment (renatgalimov) [reply anki-editor-tests.el:231]: Actually, I'm wrong.

**Review by slotThe — COMMENTED** [review]
I was about to merge this (hence the force-push, just fixing some whitespace issues, sorry!), but the test `test--note-at-point-for-examples-should-produce-correct-output` seems to fail for me

<details>
<summary>click</summary>

    F test--note-at-point-for-examples-should-produce-correct-output
        Test ‘anki-editor--note-at-point’ should produce correct output for examples.
        (ert-test-failed
         ((should (equal note-at-point expected-note)) :form
          (equal
           #s(anki-editor-note nil "Basic" "Computing"
                               (("Back"
                                 . "<div align=\"left\">\n\n<div class=\"org-src-container\">\n<pre class=\"src src-emacs-lisp\"><span style=\"color: #000000;\">(</span><span style=\"color: #4a3a8a; font-weight: bold;\">condition-case</span> the-error\n    <span style=\"color: #595959; font-style: italic;\">;; </span><span style=\"color: #595959; font-style: italic;\">the protected form</span>\n    <span style=\"color: #721045;\">(</span><span style=\"color: #4a3a8a; font-weight: bold;\">progn</span>\n      <span style=\"color: #005e8b;\">(</span>do-something-dangerous<span style=\"color: #005e8b;\">)</span>\n      <span style=\"color: #005e8b;\">(</span>do-something-more-dangerous<span style=\"color: #005e8b;\">)</span><span style=\"color: #721045;\">)</span>\n  <span style=\"color: #595959; font-style: italic;\">;; </span><span style=\"color: #595959; font-style: italic;\">handlers</span>\n  <span style=\"color: #721045;\">(</span>error-symbol1 <span style=\"color: #005e8b;\">(</span>handler1 the-error<span style=\"color: #005e8b;\">)</span><span style=\"color: #721045;\">)</span>\n  <span style=\"color: #721045;\">(</span><span style=\"color: #005e8b;\">(</span>error-symbol2 error-symbol3 <span style=\"color: #7f0000;\">(</span>handler the-error<span style=\"color: #7f0000;\">)</span><span style=\"color: #005e8b;\">)</span><span style=\"color: #721045;\">)</span><span style=\"color: #000000;\">)</span>\n</pre>\n</div>\n\n</div>\n")
                                ("Front"
                                 . "<p>\nHow to trap errors in emacs lisp?\n</p>\n"))
                               ("lisp" "emacs" "programming"))
           #s(anki-editor-note nil "Basic" "Computing"
                               (("Back"
                                 . "<div align=\"left\">\n\n<div class=\"org-src-container\">\n<pre class=\"src src-emacs-lisp\">(condition-case the-error\n    ;; the protected form\n    (progn\n      (do-something-dangerous)\n      (do-something-more-dangerous))\n  ;; handlers\n  (error-symbol1 (handler1 the-error))\n  ((error-symbol2 error-symbol3 (handler the-error))))\n</pre>\n</div>\n\n</div>\n")
                                ("Front"
                                 . "<p>\nHow to trap errors in emacs lisp?\n</p>\n"))
                               ("lisp" "emacs" "programming")))
          :value nil :explanation
          (struct-field fields
                        (list-elt 0
                                  (cdr
                                   (arrays-of-different-length 1351 345
                                                               "<div align=\"left\">\n\n<div class=\"org-src-container\">\n<pre class=\"src src-emacs-lisp\"><span style=\"color: #000000;\">(</span><span style=\"color: #4a3a8a; font-weight: bold;\">condition-case</span> the-error\n    <span style=\"color: #595959; font-style: italic;\">;; </span><span style=\"color: #595959; font-style: italic;\">the protected form</span>\n    <span style=\"color: #721045;\">(</span><span style=\"color: #4a3a8a; font-weight: bold;\">progn</span>\n      <span style=\"color: #005e8b;\">(</span>do-something-dangerous<span style=\"color: #005e8b;\">)</span>\n      <span style=\"color: #005e8b;\">(</span>do-something-more-dangerous<span style=\"color: #005e8b;\">)</span><span style=\"color: #721045;\">)</span>\n  <span style=\"color: #595959; font-style: italic;\">;; </span><span style=\"color: #595959; font-style: italic;\">handlers</span>\n  <span style=\"color: #721045;\">(</span>error-symbol1 <span style=\"color: #005e8b;\">(</span>handler1 the-error<span style=\"color: #005e8b;\">)</span><span style=\"color: #721045;\">)</span>\n  <span style=\"color: #721045;\">(</span><span style=\"color: #005e8b;\">(</span>error-symbol2 error-symbol3 <span style=\"color: #7f0000;\">(</span>handler the-error<span style=\"color: #7f0000;\">)</span><span style=\"color: #005e8b;\">)</span><span style=\"color: #721045;\">)</span><span style=\"color: #000000;\">)</span>\n</pre>\n</div>\n\n</div>\n"
                                                               "<div align=\"left\">\n\n<div class=\"org-src-container\">\n<pre class=\"src src-emacs-lisp\">(condition-case the-error\n    ;; the protected form\n    (progn\n      (do-something-dangerous)\n      (do-something-more-dangerous))\n  ;; handlers\n  (error-symbol1 (handler1 the-error))\n  ((error-symbol2 error-symbol3 (handler the-error))))\n</pre>\n</div>\n\n</div>\n"
                                                               first-mismatch-at
                                                               84))))))


</details>

**Comment by renatgalimov** [comment]
@slotThe , could it be because your org-mode source code formatting diverges from the default? 

Could you please try running the test like this?
```
emacs -batch -l ert -l anki-editor.el -l anki-editor-tests.el --eval "(let ((ert-batch-print-level 10) (ert-batch-print-length 120)) (ert-run-tests-batch-and-exit))"
```

**Comment by slotThe** [comment]
Indeed, c99e53afd531819612a7f00ae040411be9d3549a seems to fix it thanks! (FWIW, the old even failed on `emacs -Q` for me, so this is probably actually necessary.)

**Comment by orgtre** [comment]
Nice! I didn't try running the tests, I guess they should be okay, but they don't cover very much (see my comment [here](https://github.com/anki-editor/anki-editor/issues/50#issuecomment-2118836867)).

**Pull request merged**